src/main/java/bio/terra/service/tabulardata/google/BigQueryProjectProvider.java

```diff
@@ -0,0 +1,21 @@
+package bio.terra.service.tabulardata.google;
+
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Given a projectId, returns a {@link BigQueryProject} object while caching
+ */
+@Component
+public class BigQueryProjectProvider implements Function<String, BigQueryProject> {
+    private static final ConcurrentHashMap<String, BigQueryProject> BQ_PROJECT_CACHE = new ConcurrentHashMap<>();
+
+    @Override
+    public BigQueryProject apply(String projectId) {
```

Comment:
I seem to recall there were a few comments I made on this code in a previous PR, but I don't remember seeing the responses. One of the comments was regarding this change, to move the project cache out of `BigQueryProject` and wrap it in a Component. Why was this change made? If we need to support providing a prebuilt project at test time, it could be accomplished by adding a `put` API that inserts a `BigQueryProject` in the cache.

As I see it, the benefits of leaving in its original form are that the test code will be smaller, and easier to understand since it doesn't rely on DI.

Reply:
Ah shoot right.  I'm not sure that I follow that the test relies on DI.  With this it just become a parameter to the BigQueryPdao, which doesn't seem to be overly onerous.  I did mean to go in to convert the cache into not a static but a member variable.

But the change was because there didn't seem to be an easier to mock this and just moving the provider into it's own class made it easier to tell what's getting mocked in the test

Reply:
If you add a `put()` method, then there won't be a need to mock it at all. I think in general, you rarely should need to mock a a data only class.